pyproject.toml

```diff
@@ -54,6 +54,8 @@ opensearch-py = { version = "^2.3.1", optional = true }
 requests-aws4auth = { version = "1.2.3", optional = true }
 pgvector = { version = "^0.2.3", optional = true }
 psycopg2-binary = { version = "^2.9.9", optional = true }
+cattrs = {git = "https://github.com/python-attrs/cattrs.git", rev = "main"}
+marshmallow-dataclass = "^8.6.0"
```

Comment:
Leaving for now as artifacts of what else I tried. `cattrs` was somewhat promising but kept crapping out on complex classes. `marshmallow-dataclass` is exactly what we want but does not support `attrs` :(

Reply:
what kind of crap-out were you seeing?